### PR TITLE
[QE-4039] Allow Specific SDK and Core Versions to be Tested Through the Use of Flags

### PR DESCRIPTION
**Pull request by SamJHaig**
https://techweb.axway.com/jira/browse/QE-4039

### CODE REVIEW

File: Helpers/Appc_Helper.js [hunk 1]
```diff
@@ -5,11 +5,111 @@ const
...
+			Output.log('Setting Environment to PreProduction');
+			exec('appc config set defaultEnvironment preproduction');
+
+			Output.log('Logging In');
+			exec(`APPC_ENV=preproduction appc login --no-prompt --username ${appc.username} --password ${appc.password}`);
+
+			// Check if the CLI version we want to use is installed
+			Output.log(`Checking if the Latest Version of ${version} is Installed`);
+			const
+				status = exec(`appc use --prerelease | grep ${version} | tail -1`).toString(),
```
Comment (ewanharris) [inline Helpers/Appc_Helper.js]: `appc use -o json --prerelease` might be more crossplatform-y to parse this? But you have to do it in that specific order for some reason, `--prerelease -o json` will try and download a version called json

File: Helpers/Appc_Helper.js [hunk 2]
```diff
@@ -5,11 +5,112 @@ const
...
+	 * @param {String} branch - The branch version desired for testing
+	 ****************************************************************************/
+	static installSDK(branch) {
+		return new Promise((resolve, reject) => {
+			Output.info(`Installing Titanium SDK From '${branch}'... `);
+
+			let
+				sdk,
+				cmd = 'appc',
+				args = [ 'ti', 'sdk', 'install', '-b', branch, '-d', '--no-prompt', '--username', appc.username, '--password', appc.password ],
```
Comment (longton95) [inline Helpers/Appc_Helper.js]: I am unable to install a GA version of the SDK as it trying to look for a branch, not a version 

`npm run test -- -p Android -s 7.4.0.GA`

File: Helpers/Appc_Helper.js [hunk 3]
```diff
@@ -5,11 +5,112 @@ const
...
+			// Go to the pre-production environment
+			Output.log('Logging Out of the Appc CLI');
+			exec('appc logout');
+
+			Output.log('Setting Environment to PreProduction');
+			exec('appc config set defaultEnvironment preproduction');
+
+			Output.log('Logging In');
+			exec(`APPC_ENV=preproduction appc login --no-prompt --username ${appc.username} --password ${appc.password}`);
+
```
Comment (longton95) [inline Helpers/Appc_Helper.js]: I am unable to install a GA CLI if I specify if, in the past, we have used old GA CLI versions in smokes

`npm run test -- -p Android -c 7.0.6`

**Comment by longton95** [comment]
LGTM

**Pull request merged**